engine/src/main/scala/cromwell/engine/db/DataAccess.scala

```diff
@@ -649,4 +650,14 @@ trait DataAccess extends AutoCloseable {
       }
     }
   }
+
+  def refreshWorkflowMetadataSummaries(startMetadataId: Long, startMetadataTimestamp: Option[OffsetDateTime])
+                                      (implicit ec: ExecutionContext): Future[Long] = {
+    self.refreshMetadataSummaries(startMetadataId, startMetadataTimestamp)
+  }
+
+  def getWorkflowStatus(id: WorkflowId)
+                       (implicit ec: ExecutionContext): Future[Option[WorkflowState]] = {
+    self.getStatus(id) map { _ map WorkflowState.fromString }
```

Comment:
Here is where, IMHO, the translation boundary between the database and engine should exist, _not_ in the database code. Effectively, everything outside of only `case class WorkflowMetadataSummary(…)` should be up in the `engine`.